components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/UIBrandingPreferenceResolver.java

```diff
@@ -48,9 +48,33 @@ public interface UIBrandingPreferenceResolver {
      */
     void clearBrandingResolverCacheHierarchy(String currentTenantDomain) throws BrandingPreferenceMgtException;
 
+    /**
+     * This method is used to retrieve a resolved custom text preference.
+     *
+     * @param type   Type of the custom text preference.
+     * @param name   Name of the tenant/application.
+     * @param screen Screen param of the custom text preference.
+     * @param locale Language preference of the custom text preference.
+     * @return The resolved custom text preference.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
     default CustomText resolveCustomText(String type, String name, String screen, String locale)
             throws BrandingPreferenceMgtException {
 
         throw new NotImplementedException("This functionality is not implemented.");
     }
+
+    /**
+     * This method is used to clear the custom text preference resolver caches of
+     * the organization and all its children down the tree.
+     *
+     * @param currentTenantDomain   Tenant domain where the cache needs to be cleared.
+     * @param screen                Screen param of the custom text.
+     * @param locale                Locale param of the custom text.
+     * @throws BrandingPreferenceMgtException if any error occurred.
+     */
+    default void clearCustomTextResolverCacheHierarchy(String currentTenantDomain, String screen, String locale)
+            throws BrandingPreferenceMgtException {
+
```

Comment:
If we are adding this as a default method shall we throw the `NotImplementedException` as in the `resolveCustomText` method?

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/24/commits/c3cef6944aa26bd9357f85464d19749758380ca7